### PR TITLE
Add Clortho and JWKS support

### LINKED ISSUE #120: Support for JWKS
The current Resolver implementation expects endpoints to provide raw public keys.  We would like support for wrapped keys in either JWK or JWK set format by examining the content-type header of the response from the key URI.

### PR DESCRIPTION
**Pull request by renaz6**
- Remove key package
- add clortho support 
- closes #120

### CODE REVIEW

File: basculehttp/bearerTokenFactory_test.go [hunk 1]
```diff
@@ -123,18 +123,18 @@ func TestBearerTokenFactory(t *testing.T) {
...
 				token := jwt.NewWithClaims(jwt.SigningMethodHS256, tc.claims)
 				token.Valid = tc.validToken
 				p.On("ParseJWT", mock.Anything, mock.Anything, mock.Anything).Return(token, tc.parseErr).Once()
 			}
 			if tc.resolveCalled {
-				r.On("ResolveKey", mock.Anything, mock.Anything).Return(pair, tc.resolveErr).Once()
+				r.On("Resolve", mock.Anything, mock.Anything).Return(key, tc.resolveErr).Once()
 				if tc.resolveErr == nil {
-					pair.On("Public").Return(nil).Once()
+					key.On("Public").Return(nil).Once()
```
Comment (denopink) [inline basculehttp/bearerTokenFactory_test.go:137]: You'll want to add `pair.AssertExpectations(t)` and `key.AssertExpectations(t)` to ensure mock calls were made

File: basculehttp/bearerTokenFactory_test.go [hunk 2]
```diff
@@ -234,3 +234,73 @@ good:
...
+				assert.NoError(err)
+				return
+			}
+			require.Error(err)
+			assert.True(strings.Contains(err.Error(), tc.expectedErr.Error()),
+				fmt.Errorf("error [%v] doesn't contain error [%v]",
+					err, tc.expectedErr),
+			)
+		})
+	}
```
Comment (denopink) [inline basculehttp/bearerTokenFactory_test.go:305]: I've barely touched fx atm, I'll rely on @johnabass on fx tests while I learn how to use fx!!! 📚

File: go.mod [hunk 3]
```diff
@@ -15,9 +15,9 @@ require (
 	github.com/subosito/gotenv v1.4.0 // indirect
 	github.com/xmidt-org/arrange v0.3.0
 	github.com/xmidt-org/candlelight v0.0.10
+	github.com/xmidt-org/clortho v0.0.2
 	github.com/xmidt-org/sallust v0.1.6
 	github.com/xmidt-org/touchstone v0.1.1
-	github.com/xmidt-org/webpa-common v1.11.9
```
Comment (denopink) [inline go.mod:20]: nice!

File: basculehttp/bearerTokenFactory.go [hunk 4]
```diff
@@ -26,7 +26,7 @@ import (
 	"github.com/golang-jwt/jwt"
 	"github.com/xmidt-org/arrange"
 	"github.com/xmidt-org/bascule"
-	"github.com/xmidt-org/bascule/key"
+	"github.com/xmidt-org/clortho"
 	"go.uber.org/fx"
```
Comment (denopink) [inline basculehttp/bearerTokenFactory.go:30]: We're not using `bascule` anymore after this, right? If so, we should probably rename files, types and the dirs that refer to the name `bascule`.

**Review by denopink — CHANGES_REQUESTED** [review]
very cool! 🍻

File: basculehttp/bearerTokenFactory_test.go (hunk 1 above)
Comment (renaz6) [reply basculehttp/bearerTokenFactory_test.go:137]: yep, thanks for the reminder. I am still getting used to unit testing in Go. This has been updated in my most recent commit.

File: basculehttp/bearerTokenFactory_test.go [hunk 5]
```diff
@@ -234,3 +235,73 @@ good:
...
+			key:         "bad",
+			optional:    true,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.description, func(t *testing.T) {
+			result := In{}
+			assert := assert.New(t)
+			require := require.New(t)
+			app := fx.New(
```
Comment (johnabass) [inline basculehttp/bearerTokenFactory_test.go:283]: you can also use the `fxtest` package, which simplifies a lot of this kind of testing code.  `fxtest.New(...)` does a lot of setup around logging and assertions that will reduce the code here in this test.

File: basculehttp/mocks_test.go [hunk 6]
```diff
@@ -49,3 +54,52 @@ func (p *mockParser) ParseJWT(token string, claims jwt.Claims, parseFunc jwt.Key
 	_, err = parseFunc(t)
 	return t, err
 }
+
+// MockKey is a mock for key.
+type MockKey struct {
```
Comment (johnabass) [inline basculehttp/mocks_test.go:59]: There's no harm in having this be an exported type from a `_test` file.  But, in general, it's good to leave mocks as unexported, e.g. `mockKey`.  That makes it clearer to new developers that you never intend this type to be used in tests outside this package.

**Review by johnabass — COMMENTED** [review]
Looks great!  Well done, nice code.  I only had a couple of comments as food for thought.  You can change the code if you like, but it's not necessary.

File: basculehttp/mocks_test.go (hunk 6 above)
Comment (renaz6) [reply basculehttp/mocks_test.go]: Got it. Updated in my recent commit.

File: basculehttp/bearerTokenFactory_test.go (hunk 5 above)
Comment (renaz6) [reply basculehttp/bearerTokenFactory_test.go:283]: Thanks for the tip! I'm looking into the `fxtest` package but will leave the test as is for now until I completely understand the `fxtext` package

**Review by denopink — APPROVED** [review]
lgtm! 🍻

**Comment by denopink** [comment]
@renaz6 you may need to rebase with main since there were a few dependency updates.

File: basculehttp/bearerTokenFactory_test.go [hunk 7]
```diff
@@ -234,3 +233,73 @@ good:
 		})
 	}
 }
+
+// func TestProvideResolver(t *testing.T) {
```
Comment (denopink) [inline basculehttp/bearerTokenFactory_test.go:237]: We probably want to remove this

**Review by johnabass — APPROVED** [review]

**Pull request merged**